main.nf

```diff
@@ -1174,19 +1236,156 @@ process ConcatVCF {
     // we have this funny *_* pattern to avoid copying the raw calls to publishdir
         set variantCaller, idPatient, idSample, file("*_*.vcf.gz"), file("*_*.vcf.gz.tbi") into vcfConcatenated
 
-    when: 'haplotypecaller' in tools || 'mutect2' in tools || 'freebayes' in tools
+    when: ('haplotypecaller' in tools || 'mutect2' in tools || 'freebayes' in tools)
 
     script:
-    if (variantCaller == 'HaplotypeCallerGVCF') outputFile = "HaplotypeCaller_${idSample}.g.vcf"
-    else outputFile = "${variantCaller}_${idSample}.vcf"
+    if (variantCaller == 'HaplotypeCallerGVCF') 
+      outputFile = "HaplotypeCaller_${idSample}.g.vcf"
+    else if (variantCaller == "Mutect2") 
+      outputFile = "unfiltered_${variantCaller}_${idSample}.vcf"
+    else 
+      outputFile = "${variantCaller}_${idSample}.vcf"
     options = params.targetBED ? "-t ${targetBED}" : ""
     """
     concatenateVCFs.sh -i ${genomeIndex} -c ${task.cpus} -o ${outputFile} ${options}
     """
 }
 
+(vcfConcatenated, vcfConcatenatedForFilter) = vcfConcatenated.into(2)
 vcfConcatenated = vcfConcatenated.dump(tag:'VCF')
 
+process PileupSummariesForMutect2 {
+    tag {idSampleTumor + "_vs_" + idSampleNormal + "_" + intervalBed.baseName }
+
+    input:
+        set idPatient, idSampleNormal, file(bamNormal), file(baiNormal), idSampleTumor, file(bamTumor), file(baiTumor), file(intervalBed) from pairBamPileupSummaries 
+        set file(germlineResource), file(germlineResourceIndex) from Channel.value([
+            referenceMap.germlineResource,
+            referenceMap.germlineResourceIndex
+            ])
+        set idPatient,
+            idSampleNormal,
+            idSampleTumor,
+            file(statsFile) from intervalStatsFiles
+
+    output:
+        set idPatient,
+            idSampleTumor,
+            file("${intervalBed.baseName}_${idSampleTumor}_pileupsummaries.table") into pileupSummaries
+
+    when: 'mutect2' in tools && params.pon
+
+    script:
+    """
+        # pileup summaries
+        gatk --java-options "-Xmx${task.memory.toGiga()}g" \
+            GetPileupSummaries \
+            -I ${bamTumor} \
+            -V ${germlineResource} \
+            -L ${intervalBed} \
+            -O ${intervalBed.baseName}_${idSampleTumor}_pileupsummaries.table
+    """
+}
+
+pileupSummaries = pileupSummaries.groupTuple(by:[0,1])
+
+process MergePileupSummaries {
+    tag {idPatient + "_" + idSampleTumor}
+
+    publishDir "${params.outdir}/VariantCalling/${idSampleTumor}/Mutect2", mode: params.publishDirMode
+
+    input:
+        file(genomeDict) from referenceMap.genomeDict
```

Comment:
```suggestion
        file(genomeDict) from Channel.value([referenceMap.genomeDict])
```

Reply:
why to make an extra channel?

Reply:
I think the behavior is the same in this case, you'll always get a `ValueChannel` either way: The latter proposed way just states it explicitly, so even someone not being aware of this behavior of nextflow will know this is a `ValueChannel` and not a `QueueChannel` now.